pkg/apis/serving/v1alpha2/framework_custom.go

```diff
@@ -55,8 +54,5 @@ func (c *CustomSpec) Validate(config *InferenceServicesConfig) error {
 	if err != nil {
 		return fmt.Errorf("Custom container validation error: %s", err.Error())
 	}
-	if c.GetStorageUri() != "" && c.Container.Name != constants.InferenceServiceContainerName {
```

Comment:
actually this validation is really for the purpose of storage initializer injection, if the model server container name is not `InferenceServiceContainerName` the storage initializer won't get injected for custom if user intend to use it. https://github.com/kubeflow/kfserving/blob/dafd2e6997a6decd4457af62896c72f025e8c2a5/pkg/webhook/admission/pod/storage_initializer_injector.go#L96